executor/common_linux.h

```diff
@@ -1485,6 +1485,139 @@ static long syz_io_uring_submit(volatile long a0, volatile long a1, volatile lon
 
 #endif
 
+#if SYZ_EXECUTOR || __NR_syz_btf_id_by_name
+
+#include <errno.h>
+#include <fcntl.h>
+#include <linux/btf.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#define VMLINUX_MAX_SUPPORT_SIZE (10 * 1024 * 1024)
+
+// Read out all the content of /sys/kernel/btf/vmlinux to the buffer and return
+// it.
+static char* read_btf_vmlinux()
+{
+	int fd;
+	struct stat stat;
+	unsigned long vmlinux_size;
+	unsigned long bytes_read = 0;
+
+	static char buf[VMLINUX_MAX_SUPPORT_SIZE];
+
+	fd = open("/sys/kernel/btf/vmlinux", O_RDONLY);
+	if (fd < 0)
+		return NULL;
+
+	if (fstat(fd, &stat) < 0)
+		return NULL;
+	vmlinux_size = stat.st_size;
+
+	if (vmlinux_size > VMLINUX_MAX_SUPPORT_SIZE)
+		return NULL;
+
+	// Read the whole vmlinux file into buf.
+	while (bytes_read != vmlinux_size) {
+		ssize_t ret = read(fd, buf + bytes_read, vmlinux_size - bytes_read);
+		if (ret <= 0) {
+			return NULL;
+		}
+		bytes_read += ret;
+	}
+	return buf;
+}
+
+// Given a pointer to a C-string as the only argument a0, return its
+// corresponding btf ID. Fail if there is an error opening the vmlinux file or
+// the ID is not found.
+static long syz_btf_id_by_name(volatile long a0)
+{
+	// syzlang: syz_btf_id_by_name(name ptr[in, string]) btf_id
+	// C:       syz_btf_id_by_name(char* name)
+
+	char* target = (char*)a0;
```

Comment:
Do I need to verify the C string? (or else a segmentation fault might occur)

Reply:
No, no need to check it. executor handles SIGSEGVs for all pseudo-syscalls separately.